IntegrationTests/test_runner.py

```diff
@@ -0,0 +1,484 @@
+        with open(file_path, "r", encoding="utf-8") as f:
+            contents = f.read()
+            if "VM:" in contents:
+                test_files.append(str(file_path))
+
+    return test_files
+
+
+def read_directory(path, test_files):
+    """
+    Recursively read all sub directories
+    Path is the directory we are currently in
+    test_files is the list of test files we are building up
+    """
+    for directory in Path(path).iterdir():
+        if directory.is_dir():
+            read_directory(directory, test_files)
+        else:
+            continue
+
+    locate_tests(path, test_files)
+
+
+def collect_tests(test_files):
+    """
+    Assemble a dictionary of
+    name: the name of the test file
+    stdout/stderr: the expected output of the test
+    """
+    tests = []
+    for file_path in test_files:
+        test_dict = parse_test_file(file_path)
+        if test_dict is None:
+            continue
+        tests.append(test_dict)
+
+    return tests
+
+
+def parse_test_file(test_file):
+    """
+    parse the test file to extract the important information
+    """
+    test_info_dict = {
+        "name": test_file,
+        "stdout": [],
+        "stderr": [],
+        "custom_classpath": None,
+        "case_sensitive": False,
+    }
+    with open(test_file, "r", encoding="utf-8") as open_file:
+        contents = open_file.read()
+        comment = contents.split('"')[1]
+
+        # Make sure if using a custom test classpath that it is above
+        # Stdout and Stderr
+        if "custom_classpath" in comment:
+            comment_lines = comment.split("\n")
+            for line in comment_lines:
+                if "custom_classpath" in line:
+                    test_info_dict["custom_classpath"] = line.split(
+                        "custom_classpath:"
+                    )[1].strip()
+
+                    classpath_t = test_info_dict["custom_classpath"]
+
+                    # Now check our custom classpath for any tags
+                    # Tags are defined as @tag in the classpath
+                    # Will then assign the EXACT value of the
+                    # Environment variable to that spot
+
+                    if "@" in classpath_t:
+
+                        classpath_joined = ""
+                        # Does the classpath have a splitter ":"
+                        if ":" in classpath_t:
+                            split_list = classpath_t.split(":")
+                            for tag in split_list:
+                                if "@" in tag:
+                                    tag = tag.replace("@", "")
+                                    classpath_joined += os.environ[tag] + ":"
+                                    continue
+                                classpath_joined += tag + ":"
+                        else:
+                            classpath_t = classpath_t.replace("@", "")
+                            classpath_joined += os.environ[classpath_t]
```

Comment:
Same here. See for instance https://github.com/rhys-h-walker/SOMpp/actions/runs/16354125584/job/46208103214#step:9:75